x/stakeibc/keeper/community_pool.go

```diff
@@ -1,12 +1,104 @@
 package keeper
 
 import (
+	"errors"
```

Comment:
I found some of these function names a bit confusing at first glance - I'm wondering if we can improve some of them (although, I'm struggling to find good alternatives). 

I think the main source of confusion was the ICA in the name and the similarity with the transfer function. So maybe something like:
```
ICQCommunityPoolDepositICABalances → ICQCommunityPoolDepositBalance
IBCTransferCommunityPoolICATokensToStride → TransferAndStakeCommunityPoolTokens
```
wdyt?

Reply:
Yeah, I think I want to remove all the ICQ, IBC, ICA references in function names.  

ICQs would be named [NounWhichIsBeingLookedUp]Query so for example DepositSharesQuery.  Their callback would be in a file called icqcallbacks_deposit_shares.go and the function would be named DepositSharesQueryCallback

IBC doesn't really add anything, I would name these functions something like 
Transfer[AddressFrom][Tokentype]To[AddressTo]  eg. TransferDepositTokensToHolding  or at a higher level
Transfer[ZoneFrom][Tokentype]To[ZoneTo]  eg.  TransferStrideStakedTokensToCommunityPoolZone
idk... maybe it is too long and not that consistent but I do think it is clear which direction things are moving and where you expect them to end up which is important at a glance.

I think part of the problem with ICA is that we often use it to refer to the noun (account address) but sometimes use ICA to describe the commands which the tx to the foreign chain triggers.  I know I am inconsistent on this, but the rest of our code is as well.  

In a lot of the comments I make the distinction that all these functions are really doing is sending a msg which triggers the start of something, but because it is async and it is really a chain of callbacks where the rest of the action will take place, I kind of don't like function names which make it seem like those actions happen to completion inside the function (eg. TransferAndStakeCommunityPoolTokens)